examples/custom_plot_manipulation/src/main.rs

```diff
@@ -0,0 +1,123 @@
+//! This example shows how to implement custom gestures to pan and zoom in the plot
+#![cfg_attr(not(debug_assertions), windows_subsystem = "windows")] // hide console window on Windows in release
+
+use eframe::egui::{self, plot::Legend, DragValue, Event};
+use egui::{
+    plot::{Line, PlotPoints},
+    Vec2,
+};
+
+fn main() -> Result<(), eframe::Error> {
+    let options = eframe::NativeOptions::default();
+    eframe::run_native(
+        "Plot",
+        options,
+        Box::new(|_cc| Box::new(PlotExample::default())),
+    )
+}
+
+struct PlotExample {
+    lock_x: bool,
+    lock_y: bool,
+    ctrl_to_zoom: bool,
+    shift_to_horizontal: bool,
+    zoom_speed: f32,
+    scroll_speed: f32,
+}
+
+impl Default for PlotExample {
+    fn default() -> Self {
+        Self {
+            lock_x: false,
+            lock_y: false,
+            ctrl_to_zoom: false,
+            shift_to_horizontal: false,
+            zoom_speed: 1.0,
+            scroll_speed: 1.0,
+        }
+    }
+}
+
+impl eframe::App for PlotExample {
+    fn update(&mut self, ctx: &egui::Context, _: &mut eframe::Frame) {
+        egui::SidePanel::left("options").show(ctx, |ui| {
+            ui.checkbox(&mut self.lock_x, "Lock x axis");
+            ui.checkbox(&mut self.lock_y, "Lock y axis");
+            ui.checkbox(&mut self.ctrl_to_zoom, "Ctrl to zoom");
+            ui.checkbox(&mut self.shift_to_horizontal, "Shift for horizontal scroll");
+            ui.add(
+                DragValue::new(&mut self.zoom_speed)
+                    .clamp_range(0.1..=2.0)
+                    .speed(0.1),
+            );
+            ui.add(
+                DragValue::new(&mut self.scroll_speed)
+                    .clamp_range(0.1..=100.0)
+                    .speed(0.1),
+            );
+        });
+        egui::CentralPanel::default().show(ctx, |ui| {
+            let (scroll, pointer_delta, pointer_down, modifiers) = ui.input(|i| {
+                let scroll = i.events.iter().find_map(|e| match e {
+                    Event::Scroll(v) => Some(*v),
+                    _ => None,
+                });
+                (
+                    scroll,
+                    i.pointer.delta(),
+                    i.pointer.primary_down(),
+                    i.modifiers,
+                )
+            });
+
+            egui::plot::Plot::new("plot")
+                .allow_zoom(false)
+                .allow_drag(false)
+                .allow_scroll(false)
+                .legend(Legend::default())
+                .show(ui, |plot_ui| {
+                    if let Some(mut scroll) = scroll {
+                        if modifiers.ctrl == self.ctrl_to_zoom {
```

Comment:
This isn't working properly because the mouse wheel event gets converted to `Zoom` when Ctrl is down, as explained in https://github.com/emilk/egui/issues/2718. I'll see if I can draft a proposal for that issue so this one can implement the custom controls as intended.

Reply:
* https://github.com/emilk/egui/issues/2718 has been resolved now